kubernetes/INSTRUCTION.md

```diff
@@ -0,0 +1,218 @@
+```
+
+### Check logs of controller
+
+```yaml
+kubectl logs -f elastic-job-k8s-controller-6d4884c75b-z22cm -n elastic-job
+
+2020-03-19T10:13:43.532Z	INFO	controller-runtime.metrics	metrics server is starting to listen	{"addr": ":8080"}
+2020-03-19T10:13:43.534Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "elasticjob", "source": "kind source: /, Kind="}
+2020-03-19T10:13:43.534Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "elasticjob", "source": "kind source: /, Kind="}
+2020-03-19T10:13:43.534Z	INFO	controller-runtime.controller	Starting EventSource	{"controller": "elasticjob", "source": "kind source: /, Kind="}
+2020-03-19T10:13:43.534Z	INFO	setup	starting manager
+2020-03-19T10:13:43.534Z	INFO	controller-runtime.manager	starting metrics server	{"path": "/metrics"}
+2020-03-19T10:13:43.822Z	DEBUG	controller-runtime.manager.events	Normal	{"object": {"kind":"ConfigMap","namespace":"elastic-job","name":"controller-leader-election-helper","uid":"50269b8b-69ca-11ea-b995-0653198c16be","apiVersion":"v1","resourceVersion":"2107564"}, "reason": "LeaderElection", "message": "elastic-job-k8s-controller-6d4884c75b-z22cm_4cf549b7-3289-4285-8e64-647d067178bf became leader"}
+2020-03-19T10:13:44.021Z	INFO	controller-runtime.controller	Starting Controller	{"controller": "elasticjob"}
+2020-03-19T10:13:44.121Z	INFO	controller-runtime.controller	Starting workers	{"controller": "elasticjob", "worker count": 1}
+```
+
+### Deploy a ElasticJob
+ 
+Create an etcd instance and service for rdzvEndpoint.
+```
+kubectl apply -f config/samples/etcd.yaml
+``` 
+ 
+Build your own trainer image 
+
+```
+export DOCKERHUB_USER=<your_dockerhub_username>
+cd kubernetes/config/samples
+
+docker build -f $DOCKERHUB_USER/examples:0.1.0rc1 .
+docker push $DOCKERHUB_USER/examples:0.1.0rc1
+```
+
+Update `config/samples/imagenet.yaml` to use your image, scripts and rdzvEndpoint. 
+
+
+Submit the training job. 
+```yaml
+kubectl apply -f config/samples/imagenet.yaml
+```
+
+As you can see, training pod and headless services have been created.
+```yaml
+$ kubectl get pods -n elastic-job
+NAME                                          READY   STATUS    RESTARTS   AGE
+elastic-job-k8s-controller-6d4884c75b-z22cm   1/1     Running   0          11m
+imagenet-worker-0                             1/1     Running   0          5s
+imagenet-worker-1                             1/1     Running   0          5s
+
+$ kubectl get svc -n elastic-job
+NAME                TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)     AGE
+imagenet-worker-0   ClusterIP   None         <none>        10291/TCP   34s
+imagenet-worker-1   ClusterIP   None         <none>        10291/TCP   34s
+```
+
+You can adjust desired replica `.spec.replicaSpecs[Worker].replicas` and apply change to k8s. 
+```
+kubectl apply -f config/samples/imagenet.yaml
+```
+
+
+### Monitoring jobs
+
+```yaml
+kubectl describe elasticjob imagenet -n elastic-job
```

Comment:
Are all jobs in the same namespace?

Reply:
em. I think it's not generic enough here. Since our above example use namespace `elastic-job`,  we specify the namespace here. We can add a note that `Change to namespace your jobs are scheduled in.`